gitutil.py

```diff
@@ -142,8 +142,12 @@ def index_to_commit(self, message, parent_commits=None):
         The worktree remains unchanged after the method returns.
 
         @param parent_commits: optional iterable of parent commits; head is used if absent
-        @return the git.Commit object representing the newly created commit
+        @return the git.Commit object representing the newly created commit or
+        None if no changes are done
         '''
+        if not self.repo.is_dirty():
```

Comment:
not sure I agree.

firstly, the caller should rather check whether or not it makes sense to create a commit (I think?)
secondly, if the index is empty, this should imo create an empty commit (i mean, why not?)